src/Network/Algorand/Node/Api.hs

```diff
@@ -165,13 +167,15 @@ data SuggestedParams = SuggestedParams
   }
 $(deriveJSON algorandTrainOptions 'SuggestedParams)
 
+newtype TealCode = TealCode
+  { unTealCode :: ByteString
+  } deriving newtype (FromJSON, ToJSON)
 
-data TealCompileRep = TealCompileRep
-  { tcrResult :: ByteString
-  --, tcrHash :: Base32 bytes  -- why
+data TealCompilationResult = TealCompilationResult
+  { tcrHash :: Address
```

Comment:
There was a "why" comment for this field. Documentation says "base32 SHA512_256 of program bytes (Address style)". It's still not 100% obvious to me whether it should be `Address` ("Address style" is not exactly `Address`), but since it's done this way in #1, I guess we can trust it.

Reply:
I want to notice, that previous structure was deleted in first commit and then new one added in the second. So I decided to do it in the second commit